### PR TITLE
Fix for Abecedarium activity

### PR DESCRIPTION
**Pull request by ricknjacky**
fix for #870

### CODE REVIEW

**Comment by ricknjacky** [comment]
@llaske the activity doesn't seem to work if I 
>In this activity all define are load one time here and store in the Abcd global variable. I think it's better to do the same for the humane library.

**Comment by ricknjacky** [comment]
Ps:- I still haven't called the Abcd for the humane instance and yes I have included humane.js in lib and libnotify too in css

**Comment by llaske** [comment]
> the activity doesn't seem to work if I do the same for humane library

What you mean by "doesn't seem to work"?

> yes I have included humane.js in lib and libnotify too in css

It's not in your PR.

**Comment by ricknjacky** [comment]
I added it in the files locally, tried testing it, the activity didn't load up is what I meant, hence I didn't push the commit 
sorry for not being specific

**Comment by ricknjacky** [comment]
I have added the humane and libnotify files, also stored it in the Abcd global variable.

**Comment by ricknjacky** [comment]
In the previous commit i added a comment , because that was causing the activity to not load in my local machine, Can someone please guide me, as to how I should call the humane which is stored in the Abcd variable? 

image

**Comment by llaske** [comment]
You didn't include link to the `libnotify.css` file in `index.html`.
Plus `l10n` is not defined in this context. You probably have to define it locally like [here](https://github.com/llaske/sugarizer/blob/dev/activities/Abecedarium.activity/play.js#L316).

File: activities/Abecedarium.activity/play.js [hunk 1]
```diff
@@ -313,7 +313,7 @@ enyo.kind({
 		this.$.colorBar.removeClass("themeColor"+this.theme);
 		this.$.colorBar.addClass("themeColor-1");
 		var playMode = this;
-		requirejs(["webL10n"], function(webL10n) {
+		requirejs(["webL10n", "l10n"], function(webL10n,l10n) {
```
Comment (llaske) [inline activities/Abecedarium.activity/play.js:316]: I don't understand the idea here?

**Review by llaske — CHANGES_REQUESTED** [review]
It doesn't fix the issue. 
You should define `l10n` around the humane log like I mention in my previous comment.

File: activities/Abecedarium.activity/play.js (hunk 1 above)
Comment (ricknjacky) [reply activities/Abecedarium.activity/play.js:316]: I actually did the change(took it literally,sorry) that you suggested in the previous commit, will rectify it soon

**Comment by ricknjacky** [comment]
The activity doesn't work for some reason still

I have, 
>define l10n around the humane log

done this

**Comment by llaske** [comment]
Did you test it? It doesn't work. Worse: it's no longer possible to export anything :disappointed: 
I think you're going in a wrong direction: the message should be add the message [here](https://github.com/llaske/sugarizer/blob/dev/activities/Abecedarium.activity/entry.js#L126). It's where the data is stored.

**Comment by ricknjacky** [comment]
image

sorry for dragging this issue for so long, but i just cant find a way out here 😔
i have attached a screenshot in that i have called the l10n by just defiing humane and l10n and i also tested using Abcd.humane.log() and it just doesn't fix 😥 

please can you tell me where exactly, i'm getting it all wrong? i really want to know my mistake

**Comment by llaske** [comment]
`define` is an async function, `exportToImage` is expected to have a sync answer.
It's probably the reason of your issue. You should call the `define` inside the `Abcd.datastore.create`.

**Comment by ricknjacky** [comment]
@llaske Extremely sorry for late reply, I just missed out on this one. I rebased all my commits and I ran the activity again, It occurred to me today that I actually had solved this long time back and I have to select a card after clicking on save image/sound button in the activity, after testing it in that manner ig I have solved the issue. I await your feedback on this :)

**Comment by llaske** [comment]
Nice. Thanks.

**Pull request merged**